Mage.Sets/src/mage/cards/m/MiragePhalanx.java

```diff
@@ -0,0 +1,101 @@
+import mage.target.targetpointer.FixedTargets;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class MiragePhalanx extends CardImpl {
+
+    private static final String ruleText =
+            "As long as {this} is paired with another creature, each of those creatures has " +
+                    "\"At the beginning of combat on your turn, create a token that’s a copy of this creature, " +
+                    "except it has haste and loses soulbond. " +
+                    "Exile it at end of combat.\"";
+
+    public MiragePhalanx(UUID ownderId, CardSetInfo setInfo) {
+        super(ownderId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{R}{R}");
+
+        this.addSubType(SubType.HUMAN);
+        this.addSubType(SubType.SOLDIER);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Soulbond
+        this.addAbility(new SoulbondAbility());
+
+        // As long as Mirage Phalanx is paired with another creature, each of those creatures has
+        // “At the beginning of combat on your turn, create a token that’s a copy of this creature,
+        // except it has haste and loses soulbond.
+        // Exile it at end of combat.”
+        Ability ability = new BeginningOfCombatTriggeredAbility(new MiragePhalanxEffect(), TargetController.YOU, false);
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityPairedEffect(ability, ruleText)));
+    }
+
+    private MiragePhalanx(final MiragePhalanx card) {
+        super(card);
+    }
+
+    @Override
+    public MiragePhalanx copy() {
+        return new MiragePhalanx(this);
+    }
+}
+
+class MiragePhalanxEffect extends OneShotEffect {
+    MiragePhalanxEffect() {
+        super(Outcome.PutCreatureInPlay);
+        this.staticText = "create a token that’s a copy of this creature, " +
+                "except it has haste and loses soulbond. " +
+                "Exile it at end of combat.";
+    }
+
+    private MiragePhalanxEffect(final MiragePhalanxEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = source.getSourcePermanentOrLKI(game);
+        if (permanent == null) { return false; }
+
+        // Create the token, it has haste
+        CreateTokenCopyTargetEffect tokenCopyEffect = new CreateTokenCopyTargetEffect(source.getControllerId(), null, true);
+        tokenCopyEffect.setTargetPointer(new FixedTarget(permanent, game));
+        // It loses soulbond
+        tokenCopyEffect.addAbilityClassesToRemoveFromTokens(SoulbondAbility.class);
+
+        tokenCopyEffect.apply(game, source);
+
+        if (tokenCopyEffect.getAddedPermanents().isEmpty()) { return false; }
+
+        // Exile it at the end of combat
+        ExileTargetEffect exileEffect = new ExileTargetEffect("Exile it");
+        exileEffect.setTargetPointer(new FixedTargets(tokenCopyEffect.getAddedPermanents(), game));
```

Comment:
RE: exiling them off a single trigger. I currently am using FixedTargets. They still show up on the stack individually, letting me cast an instant or activate an ability in-between each exile effect.

Reply:
Also, I don't think those arrows are pointing correctly (neither of the two creatures being pointed at are the tokens, those are the originals.
<img width="1385" alt="Screen Shot 2022-02-05 at 7 24 20 PM" src="https://user-images.githubusercontent.com/48962821/152663440-8726cc70-e6e2-443e-8274-e83397b7d88f.png">

Reply:
You only need to create a single trigger and have that one target all the tokens.

Reply:
Ahh. I see my confusion. I'm doing it inside of the MiragePhalanxEffect. So I will make one per creature since each one has that effect. Hmm...

Reply:
Wait, I think there was a miscommunication. There should be a separate trigger for the Phalanx and the creature it's copying. I was referring to when there's a Doubling Season and the trigger makes multiple of the same token.

Reply:
I see. Okay, that makes more sense. So the code is currently using FixedTargets. One triggers goes on the stack for Phalanx and one for the other card (First-Wing in this case). 

I tried it out with doubling season on the board. It's only a single trigger stack on the stack for each of them, but it destroys both tokens (Both Phalanx tokens off Phalanx trigger and both First-Wing tokens off the First-Wing trigger).

Reply:
Perfect, that's what should be happening.